report/src/main/webapp/app/services/generatePdfDraw.service.js

```diff
@@ -0,0 +1,197 @@
+/*
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(['angularAMD', 'endpointConfiguration', 'generatePdfDataService'], function (angularAMD) {
+  'use strict';
+  angularAMD.factory('generatePdfDrawService', generatePdfDrawService);
+
+  /**
+   * Service responsible for drawing pdf report
+   */
+  function generatePdfDrawService(generatePdfDataService) {
+    var service = {
+      generateDocHeader: generateDocHeader,
+      generateChart: generateChart,
+      generateChartsForTests: generateChartsForTests,
+    };
+
+    return service;
+
+    function generateDocHeader(ctx) {
+      var projectName = $('.toolbar-blocks>.toolbar-block:nth-child(3)>.name').text();
+      var suiteName = $('.toolbar-blocks>.toolbar-block:nth-child(4)>.name').text();
+      var suiteVersion = $('.toolbar-blocks>.toolbar-block:nth-child(4)>.preformatted').text();
+      var runDate = $('.toolbar-blocks>.toolbar-block:nth-child(5)').text().trim();
+
+      var urlString = window.location.href;
+      var url = new URL(urlString);
+      var company = url.searchParams.get('company');
+      var project = url.searchParams.get('project');
+      var correlationId = $('.toolbar-blocks>.toolbar-block:first-child>span').text().trim();
+      var domain = urlString.split('report.html')[0];
+      var testUrl = domain + 'report.html?company=' + company + '&project=' + project + '&correlationId=' + correlationId;
```

Comment:
Maybe *testUrl* could be created by some service that already have information about `company`, `project` and `correlationId` ?